In [1]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import BaggingClassifier

from sklearn.metrics import accuracy_score
from sklearn import datasets
from sklearn.model_selection import train_test_split

import warnings
warnings.filterwarnings("ignore")


C:\Users\175045\Anaconda3\lib\importlib\_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)


In [2]:
brc = datasets.load_breast_cancer()

In [3]:
y = brc.target
x = brc.data

In [4]:
x.shape, y.shape

((569, 30), (569,))

In [5]:
xtrain, xtest, ytrain, ytest = train_test_split(x,y, random_state=9)

In [6]:
xtrain.shape, xtest.shape, ytrain.shape, ytest.shape

((426, 30), (143, 30), (426,), (143,))

In [8]:
dt = DecisionTreeClassifier()

# Let's check the accuracy of individual classifier.

In [10]:
dt.fit(xtrain,ytrain)
ypred = dt.predict(xtest)
accuracy_score(ytest, ypred)


0.9440559440559441

# Bagging Classifier

In [11]:
bgc = BaggingClassifier(
    DecisionTreeClassifier(),
    n_estimators=500,
    max_samples=100,
    max_features=1.0,
    bootstrap=True,
    bootstrap_features=False,
    oob_score=False,
    warm_start=False,
    n_jobs=-1,
    random_state=9,
    verbose=0,
)

# Let's check the accuracy of Individual and the Bagging classifier

In [13]:
for clf in (dt, bgc):
    clf.fit(xtrain, ytrain)
    ypred = clf. predict(xtest)
    print(clf.__class__.__name__, accuracy_score(ytest, ypred))

DecisionTreeClassifier 0.9230769230769231
BaggingClassifier 0.972027972027972


We have achived 97.2% accuracy using the Bagging Ensemble Learning

# Out-of-Bag Evaluation

In [19]:
oobaggc = BaggingClassifier(
    DecisionTreeClassifier(),
    n_estimators=500,
    max_samples=100,
    max_features=1.0,
    bootstrap=True,
    bootstrap_features=False,
    oob_score=True,
    warm_start=False,
    n_jobs=-1,
    random_state=9,
    verbose=0,
)

In [23]:
oobaggc.fit(x,y)

BaggingClassifier(base_estimator=DecisionTreeClassifier(ccp_alpha=0.0,
                                                        class_weight=None,
                                                        criterion='gini',
                                                        max_depth=None,
                                                        max_features=None,
                                                        max_leaf_nodes=None,
                                                        min_impurity_decrease=0.0,
                                                        min_impurity_split=None,
                                                        min_samples_leaf=1,
                                                        min_samples_split=2,
                                                        min_weight_fraction_leaf=0.0,
                                                        presort='deprecated',
                                                        random_state=None,


# Let's check the accuracy of Bagging classifier on Out-of-Bag data set

In [24]:
oobaggc.oob_score

True

In [25]:
oobaggc.oob_score_

0.9525483304042179